### ISSUE TITLE
MicStream.StartRecording throws an exception in the MicrophoneStream demo scene

### ISSUE THREAD

**Issue by markstacey** [Microsoft/HoloToolkit-Unity#574]
There is an error in the MicrophoneStream demo when you start recording.

The demo scene builds and works when you first run it streaming from the Mic which you can hear, but if you then go into the Awake function and comment the MicStartStream line and swap it with a MicStartRecording call like below:

if (AutomaticallyStartStream == true)
{
   //CheckForErrorOnCall(MicStream.MicStartStream(KeepAllData, false));
   CheckForErrorOnCall(MicStream.MicStartRecording(SaveFileName, false));
}
You get the following error:
SEHException: External component has thrown an exception.
at HoloToolkit.Unity.InputModule.MicStream.MicStartRecording(String filename, Boolean previewOnDevice)
at HoloToolkit.Unity.InputModule.Tests.MicStreamDemo.Awake()
at HoloToolkit.Unity.InputModule.Tests.MicStreamDemo.$Invoke0(Int64 instance, Int64* args)
at UnityEngine.Internal.$MethodUtility.InvokeMethod(Int64 instance, Int64* args, IntPtr method) 
(Filename: Line: 0)

You still get this error even if you defer the MicStartRecording call until the Update function.

Triggering the microphone recording works when running the demo scene in the editor on PC. This saves recordings to Username/Music/MicStreamRecordings, this issue seems to be specifically when built and run on HoloLens.

**Comment by StephenHodgson** [comment]
Hi, thanks for reporting this. Could you please try my fixes in https://github.com/Microsoft/HoloToolkit-Unity/pull/533. I'd like to know if it's still broken in this PR as well

**Comment by markstacey** [comment]
Hi, thanks for the quick reply.
It looks to me like commit 5ce8541ae0 (which I believe is the last one in 533) still has the same issue. I grabbed that commit, built the MicrophoneStream demo and got a bluetooth keyboard paired with the HoloLens so I didn't have to change the example code at all, when I pressed A to record followed by S to stop I got the following in the log:
SEHException: External component has thrown an exception.
   at HoloToolkit.Unity.InputModule.MicStream.MicStartRecording(String filename, Boolean previewOnDevice)
   at HoloToolkit.Unity.InputModule.Tests.MicStreamDemo.Update()
   at HoloToolkit.Unity.InputModule.Tests.MicStreamDemo.$Invoke14(Int64 instance, Int64* args)
   at UnityEngine.Internal.$MethodUtility.InvokeMethod(Int64 instance, Int64* args, IntPtr method) 
(Filename: <Unknown> Line: 0)

Saved microphone audio to YouArentRecordingRightNow
